Dockerfile

```diff
@@ -0,0 +1,12 @@
+FROM curlimages/curl AS phar-downloader
+
+ARG PIE_VERSION
+
+RUN curl -L --output /tmp/pie https://github.com/php/pie/releases/download/${PIE_VERSION}/pie.phar \
```

Comment:
Rather than downloading from GH, could we not use an artifact from the build step perhaps? :thinking:

Reply:
I've added steps for storing and retrieving artifacts, I'll test it soon on my fork.

Reply:
It works 👍😁. You can see it [here](https://github.com/Wirone/php-pie/actions/runs/11969197461) - artifact is created in `release-phar` job, then restored in `Docker binary-only image`. Image contains only PHAR now, I made it executable (with a `@TODO` waiting for [this](https://github.com/moby/buildkit/pull/5380) to be released).

<img width="1139" alt="image" src="https://github.com/user-attachments/assets/980006de-65dc-4857-a898-f52c13a49712">